### PR TITLE
Skip processing files with duplicates metrics

### LINKED ISSUE #448: inconsistent label dimensions in multiple textfiles causes HTTP 500
http://localhost:/metrics returns HTTP 500 internal server error, when there are two textfiles with same metric with different labels

Expected behaviour:
* wmi_exporter should export both metrics
* with node_exporter I have no problems to do similar things
* According to https://github.com/prometheus/client_golang/commit/c06fb788be8a05442219295095ee0e51523802f0 this should be possible 

Steps to reproduce:
1. Create `file1.prom` with `label_a`:
```
# HELP foo bar
# TYPE foo gauge
foo{label_a = "bar"} 1
```
2. Create `file2.prom` with `label_b`:
```
# HELP foo bar
# TYPE foo gauge
foo{label_b = "bar"} 1
```
3. Try to scrape it via textfile collector http://localhost:/metrics

### PR DESCRIPTION
**Pull request by breed808**
This change affects processing of single files with duplicate metrics.
Single files with duplicate metrics will be skipped, with no metrics being
collected and exposed by the exporter.

Previous duplicate metric processing across multiple files remains
unchanged (all files skipped, collector returns error).

Resolves #448

### CODE REVIEW

**Comment by breed808** [comment]
The `goversioninfo` tool is experiencing issues in the `build` step. Will need to fix that.

**Comment by carlpett** [comment]
So this would work with a rebase on the CI fixes, I assume? Change looks good, although I wonder if that file is gofmt:ed? The diff view has really weird indentation that makes it a bit hard to follow.

**Comment by breed808** [comment]
Rebase has fixed the CI. The file has been `fmt`ed; I've manually run `fmt` on the file locally and there were no proposed changes :man_shrugging:

**Review by carlpett — APPROVED** [review]

**Comment by carlpett** [comment]
Nice!
Re formatting, I seem to be seeing this on multiple PRs, maybe Github is doing something weird for my OS/browser combo :shrug:
